pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+                    if prop.name not in cc.properties:
+                        cc.properties[prop.name] = prop
+                for meth in super_.methods.values():
+                    if meth.name not in cc.methods:
+                        cc.methods[meth.name] = meth
+
+        if 'PropertyList' in params:
+            self._filter_properties(cc, params['PropertyList'])
+
+        if not params['IncludeQualifiers']:
+            self._remove_qualifiers(cc)
+
+        if not params['IncludeClassOrigin']:
+            self._remove_classorigin(cc)
+
+        return cc
+
+    def _get_instance(self, iname, namespace, **params):
+        """
+        Local method implements getinstance. This is generally used by
+        other instance methods that need to get an instance from the
+        repository.
+
+        It attempts to get the instance, copies it, and filters it
+        for input parameters like localonly, includequalifiers, and
+        propertylist.
+        """
+        try:
+            insts = self.instances[namespace]
+        except KeyError:
+            ce = CIMError(CIM_ERR_NOT_FOUND,
+                          'Namespace %s not found' % namespace)
+            raise ce
+
+        rtn_inst = None
+        for inst in insts:
+            if iname == inst.path:
+                if rtn_inst is not None:
+                    raise CIMError(CIM_ERR_FAILED, 'Invalid Repository. '
+                                   'Multiple instances with same path %s'
+                                   % rtn_inst.path)
+                rtn_inst = (copy.deepcopy(inst))
+                # TODO confirm that we do this test on create and we
+                # can stop looking through whole list.
+
+        if not rtn_inst:
+            raise CIMError(CIM_ERR_NOT_FOUND, 'Instance %s not found in '
+                                              'repository' % iname)
+
+        if 'PropertyList' in params:
+            property_list = params['PropertyList']
+            if property_list == "":
+                rtn_inst.properties = NocaseDict()
+            elif property_list is None:
+                pass
+            elif isinstance(property_list, list):
+                new_properties = NocaseDict()
+                properties = rtn_inst.properties
+                for pn in property_list:
+                    if not isinstance(pn, six.string_types):
+                        raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                                       'Property name %s in  '
+                                       'PropertyList %s invalid'
+                                       % (pn, property_list))
+
+                    if pn in properties:
+                        new_properties[pn] = properties[pn]
+                rtn_inst.properties = new_properties
+            else:
+                raise CIMError(CIM_ERR_INVALID_PARAMETER,
+                               'propertyList %s invalid' %
+                               property_list)
```

Comment:
This check is about the type of the PropertyList parameter, so I'd like to see a `type(property_list)` in the message. The entire  PropertyList parameter does not need to be shown in the message, IMO. Showing it with `%s` may not even show the type.

Reply:
This went away in creating simpler property list processor